### PR TITLE
Add env vars

### PR DESCRIPTION
**Pull request by barrachri**
With this PR you can specify for both workers and scheduler an IP:PORT using env vars `DASK_SCHEDULER_IP` and `DASK_SCHEDULER_PORT`.

https://github.com/dask/distributed/issues/1391

### CODE REVIEW

File: distributed/cli/dask_scheduler.py [hunk 1]
```diff
@@ -111,6 +112,15 @@ def del_pid_file():
         limit = max(soft, hard // 2)
         resource.setrlimit(resource.RLIMIT_NOFILE, (limit, hard))
 
+    # check if DASK_SCHEDULER_IP and DASK_SCHEDULER_PORT are defined
+    if not host:
+        dict_view = config.keys
+        if PY2:
+            dict_view = config.viewkeys
+        if {'scheduler-ip', 'scheduler-port'} <= dict_view():
```
Comment (mrocklin) [inline distributed/cli/dask_scheduler.py]: The above seems more complex than it needs to be.  If I were to write this I would say `if 'scheduler-ip' in config and 'scheduler-port' in config:`

File: distributed/cli/dask_worker.py [hunk 2]
```diff
@@ -191,8 +191,11 @@ def del_pid_file():
         t = Worker
 
     if not scheduler and not scheduler_file:
-        raise ValueError("Need to provide scheduler address like\n"
-                         "dask-worker SCHEDULER_ADDRESS:8786")
+        try:
+            scheduler = ":".join((config['scheduler-ip'], config['scheduler-port']))
+        except KeyError:
+            raise ValueError("Need to provide scheduler address like\n"
+            "dask-worker SCHEDULER_ADDRESS:8786 or with environment variables")
```
Comment (mrocklin) [inline distributed/cli/dask_worker.py]: If we're going to list environment variables as an option then we probably need to explain them.  In the interests of brevity I recommend just avoiding this change.

File: distributed/cli/tests/test_dask_worker.py [hunk 3]
```diff
@@ -165,3 +166,15 @@ def func(dask_worker):
...
+def test_scheduler_with_env_vars(loop):
+    envs = os.environ.copy()
+    ip = envs['DASK_SCHEDULER_IP'] = '127.0.0.1'
+    port = envs['DASK_SCHEDULER_PORT'] = '8786'
+    address = "tcp://{}:{}".format(ip, port).encode('utf-8')
+    with popen(['dask-scheduler', '--no-bokeh']) as sched:
+        with popen(['dask-worker',
+                    '--no-bokeh'], env=envs) as worker:
+                    assert any(address in worker.stderr.readline()
+                       for i in range(15))
```
Comment (mrocklin) [inline distributed/cli/tests/test_dask_worker.py]: The indentation here is odd.

File: distributed/config.py [hunk 4]
```diff
@@ -64,6 +64,12 @@ def load_config_file(config, path):
 
 
 def load_env_vars(config):
+    """
+    Scan environment variables for `DASK_`.
```
Comment (mrocklin) [inline distributed/config.py]: See http://dask.pydata.org/en/latest/develop.html#docstrings for style conventions

**Comment by barrachri** [comment]
I was checking Spark and it seems they still do in the [old way](https://github.com/apache/spark/blob/master/conf/spark-env.sh.template).

Did you think about naming the vars in this way?

`DASK_SCHEDULER_LISTEN_ADDRESS`
`DASK_SCHEDULER_CONTACT_ADDRESS`

**Comment by mrocklin** [comment]
Sorry for the delay in response.  In general this looks good to me.  

> Did you think about naming the vars in this way?

>     DASK_SCHEDULER_LISTEN_ADDRESS
>     DASK_SCHEDULER_CONTACT_ADDRESS

Yes.   More generally it seems that some groups prefer to specify keywords with environment variables.  Great, we should support that.  Now how do we go about supporting this generally across all command line arguments.   Presumably we should have a convention like `DASK_SCHEDULER_PORT` -> `--port` or `DASK_PORT` -> `--port`.  Is there a way to do this for all command line arguments that isn't particularly annoying or are we best off with just a brute force approach?  Are there pre-existing conventions for this sort of thing?

**Comment by barrachri** [comment]
About the names for the variables, I prefer something more explicit like `DASK_SCHEDULER_PORT` instead of `DASK_PORT`.

I'll give a deeper look inside click, but I think you can resolve this only in the boring way....
In case, would it be a new PR or you prefer to add new commits to this one?

**Comment by mrocklin** [comment]
The same PR would be fine to me.  Thank you for taking this on.

**Comment by barrachri** [comment]
Sorry for the delay.

I didn't find an easier solution than the one inside the last commit.

With `click` you can do something like this
```python
@click.command()
@click.option('--host', type=str, default='', help="URI, IP or hostname of this server")
@click.pass_context
def cli(ctx, debug):
    print(ctx.params) #dict
```
and you will get a dictionary, but you will miss the specs about types, so you will still have to hardcode when you cast from str to another type.

**Comment by jakirkham** [comment]
Found myself reaching for this functionality recently. Agree it is desirable. ;)

FWIW Click appears to have a way to take [environment variable for options]( http://click.pocoo.org/5/options/#values-from-environment-variables ). Should point out this makes prefixing trivial thanks to `auto_envvar_prefix` and works exactly as imagined. :)

**Comment by mrocklin** [comment]
This PR has gone stale.  Closing for now.

**Pull request closed without merging**